src/main/kotlin/org/rust/ide/annotator/RsExpressionAnnotator.kt

```diff
@@ -59,9 +59,12 @@ class RsExpressionAnnotator : Annotator {
         } else {
             if (missingFields.isNotEmpty()) {
                 val structNameRange = expr.parent.childOfType<RsPath>()?.textRange
-                if (structNameRange != null)
+                if (structNameRange != null) {
                     holder.createErrorAnnotation(structNameRange, "Some fields are missing")
                         .registerFix(AddStructFieldsFix(decl.namedFields, missingFields, expr), expr.parent.textRange)
+                    holder.createErrorAnnotation(structNameRange, "Some fields are missing")
```

Comment:
Just noticed this one. You don't need to created multiple annotations to add multiple fixed. Just extract `Annotation` to a variable and then call `registerFix` on it as much as you need.

```kotlin
val ann = holder.createErrorAnnotation(structNameRange, "Some fields are missing")
ann.registerFix()
ann.registerFix()
```